File: Contests/2019-2020/Bronze/Jan/photo.cpp
#include <bits/stdc++.h>
#pragma GCC optimize(2)
using namespace std;
typedef long long ll;
const int MAXN = 1005;

int b[MAXN], n, flag;
int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    freopen("photo.in", "r", stdin);
    freopen("photo.out", "w", stdout);
    
    cin >> n;
    for (int i = 1; i < n; ++i) cin >> b[i];
    for (int i = 1; i <= n; ++i) {
        vector<int> a(n);
        a[0] = i;
        flag = 1;
        for (int j = 1; j < n; ++j) {
            a[j] = b[j] - a[j-1];
            if (a[j]<1 || count(a.begin(), a.end(), a[j]) > 1) {
                flag = 0;
                break;
            }
        }
        if (flag) {
            for (int i = 0; i < a.size(); ++i) {
                if (i) cout << " " << a[i];
                else cout << a[i];
            }
            break;
        }
    }
    return 0;
}

File: Contests/2019-2020/Bronze/Jan/photo.py
import sys
sys.stdin = open("photo.in", "r")
sys.stdout = open("photo.out", "w")

n = int(input())
b = [0] + [int(i) for i in input().split()]
for i in range(1, n+1):
    a = [0] * n
    a[0] = i
    flag = 1
    for j in range(1, n):
        a[j] = b[j] - a[j-1]
        if a[j]<1 or a.count(j)>1:
            flag = 0
            break
    if flag:
        for j in range(n-1):
            print(a[j], end = ' ')
        print(a[n-1])
        break